v2/pkg/pool/limitbufferpool.go

```diff
@@ -0,0 +1,97 @@
+package pool
+
+import (
+	"bytes"
+	"context"
+	"runtime"
+	"time"
+)
+
+// LimitBufferPool is a pool of buffers that is limited in size and is limiting the max size of buffers that should be recycled
+// In addition, it runs a GC runtime that randomly resets a buffer every second to keep the memory usage low when usage is low
+// This is an alternative to sync.Pool, which can grow unbounded rather quickly
+type LimitBufferPool struct {
```

Comment:
Please add a few tests that ensure the correctness and efficiency of the library. Testing this in an integration test will be hard without automation.

Reply:
added

Reply:
Did you forget to push?